cpp/src/copying/copy.cu

```diff
@@ -114,6 +143,125 @@ struct copy_if_else_functor_impl<string_view> {
+
+ private:
+  Predicate _pred;
+  size_type _null_index;
+};
+
+/**
+ * @brief Adapter to negate predicates.
+ */
+template <typename Predicate>
+class logical_not {
+ public:
+  explicit logical_not(Predicate predicate) : _pred{predicate} {}
+
+  bool __device__ operator()(size_type i) const { return not _pred(i); }
+
+ private:
+  Predicate _pred;
+};
+
+/**
+ * @brief Implementation of copy_if_else() with gather()/scatter().
+ *
+ * Currently supports only nested-type column_views. Scalars are not supported.
+ */
+template <typename Left,
+          typename Right,
+          typename Filter,
+          CUDF_ENABLE_IF(not std::is_same<Left, cudf::scalar>::value &&
+                         not std::is_same<Right, cudf::scalar>::value)>
+std::unique_ptr<column> scatter_gather_based_if_else(Left const& lhs,
+                                                     Right const& rhs,
+                                                     size_type size,
+                                                     Filter is_left,
+                                                     rmm::cuda_stream_view stream,
+                                                     rmm::mr::device_memory_resource* mr)
+{
+  auto null_map_entry = size + 1;  // Out of bounds index, for gather() to nullify.
+
+  auto gather_lhs = make_counting_transform_iterator(
+    size_type{0}, lhs_gather_map_functor<Filter>{is_left, null_map_entry});
+
+  auto lhs_gathered_columns = cudf::detail::gather(table_view{std::vector<cudf::column_view>{lhs}},
+                                                   gather_lhs,
+                                                   gather_lhs + size,
+                                                   out_of_bounds_policy::NULLIFY,
+                                                   stream,
+                                                   mr)
+                                ->release();
+  auto& lhs_partial_output = lhs_gathered_columns[0];
+
+  auto scatter_map_rhs = rmm::device_uvector<size_type>{static_cast<std::size_t>(size), stream};
+  auto scatter_map_end = thrust::copy_if(rmm::exec_policy(stream),
+                                         thrust::make_counting_iterator(size_type{0}),
+                                         thrust::make_counting_iterator(size_type{size}),
+                                         scatter_map_rhs.begin(),
+                                         logical_not{is_left});
+
+  auto scatter_src_rhs = cudf::detail::gather(table_view{std::vector<column_view>{rhs}},
+                                              scatter_map_rhs.begin(),
+                                              scatter_map_end,
+                                              out_of_bounds_policy::DONT_CHECK,
+                                              stream);
+
+  auto result = cudf::detail::scatter(
+    table_view{std::vector<column_view>{scatter_src_rhs->release()[0]->view()}},
```

Comment:
This seems dangerous.  
```suggestion
    table_view{std::vector<column_view>{scatter_src_rhs->get_column(0)->view()}},
```

Reply:
Hmm... It isn't, though: the `vector<unique_ptr<>>` should be "alive" as long the expression in which it's defined is "alive".

That said, your way is far more preferable: the behaviour is unambiguous, and the fetch of the column is explicit. 

Thanks for pointing this out. I'll make this change very shortly.

Reply:
Fixed.